web/src/server/api/routers/traces.ts

```diff
@@ -49,6 +55,76 @@ export type ObservationReturnType = Omit<
   traceId: string;
 };
 
+export type TracesAllReturnType = {
+  id: string;
+  timestamp: Date;
+  name: string | undefined;
+  projectId: string;
+  userId: string | undefined;
+  release: string | undefined;
+  version: string | undefined;
+  public: boolean;
+  bookmarked: boolean;
+  sessionId: string | undefined;
+  tags: string[];
+};
+
+export const convertToReturnType = (
+  row: TracesTableReturnType,
+): TracesAllReturnType => {
+  return {
+    id: row.id,
+    name: row.name,
+    timestamp: new Date(row.timestamp),
+    tags: row.tags,
+    bookmarked: row.bookmarked,
+    release: row.release,
+    version: row.version,
+    projectId: row.project_id,
+    userId: row.user_id,
+    sessionId: row.session_id,
+    public: row.public,
+  };
+};
```

Comment:
Shouldn't those go into a traces repository?

Reply:
If we want to do this, we would need all the types to shared. I would prefer to keep them where they are and maybe move them to a dedicated file.